File: scripts/python/temp.py
#!/usr/bin/env python

import glob

devicelist = glob.glob('/sys/bus/w1/devices/28*')

if len(devicelist) > 0:
    devicefile = devicelist[0] + '/w1_slave'

class Temperature (object):

    def read(self):

        if devicefile is None:
            return None

        with open(devicefile, 'r') as f:
            lines = f.readlines()
            status = lines[0][-4:-1]
            if status == "YES":
               tempstr = lines[1][-6:-1]
               return float(tempstr) / 1000

        return None

def main():
    temp = Temperature()
    print "Device File: %s" % (devicefile)
    print "Temperature: %.3f" % (temp.read())

if __name__ == "__main__":
    main()

File: node/util/temp.js
var sh = require('shelljs');

var deviceList = '/sys/bus/w1/devices/28*';
var deviceName = '/w1_slave';
var fileReadOptions = { encoding: 'utf8' };
var deviceFile = sh.ls(deviceList).shift();

module.exports = {
    read: function() {
        if (deviceFile) {
            var fileName = deviceFile + deviceName;
            var data = fs.readFileSync(fileName, fileReadOptions).split('\n');
            if (data[0].split(' ').pop() == 'YES') {
                var temp = data[1].split('=').pop();
                return parseFloat(temp) / 1000;
            }
        }
        return null;
    }
}